osprey_flask_app/routes.py

```diff
@@ -44,17 +46,51 @@ def osprey_route():
     """
     args = request.args
     arg_dict = create_full_arg_dict(args)
-    outpath = run_full_rvic(arg_dict)
+    validate_response = validate_inputs(arg_dict)
+    if validate_response.status_code == 400:
+        return validate_response
+
+    rvic_thread = threading.Thread(
+        target=lambda q, arg: q.put(run_full_rvic(arg)), args=(que, arg_dict)
+    )
+    rvic_thread.start()
+    return Response(
+        "RVIC Process started. Check status: "
+        + url_for("osprey.status_route", thread_id=rvic_thread.native_id),
+        status=202,
+    )
+
+
+@osprey.route("/status/<thread_id>", methods=["GET"])
+def status_route(thread_id):
+    """Provide route to check status of RVIC process."""
+    active_thread_ids = [str(t.native_id) for t in threading.enumerate()]
+    if thread_id in active_thread_ids:
+        return Response("Process is still running.", status=201)
```

Comment:
This is a great start. Question though: do we have any way to check the progress from the job? If so, it would be helpful to give a "percentage complete". But we can delegate that to a future issue/PR.

Reply:
The closest thing we have for progress checking is that the `FullRVIC` class has [percentage points at certain stages of the process](https://github.com/pacificclimate/osprey/blob/84b1d0b8890737c469a999119bdc8246a32b8d5b/osprey/processes/wps_full_rvic.py#L34). If I can figure out how to return those while the job's executing, we could start with that.

Reply:
Yeah, that's maybe better than nothing.